clang/lib/CodeGen/CodeGenTBAA.cpp

```diff
@@ -95,8 +95,6 @@ static bool TypeHasMayAlias(QualType QTy) {
 
 /// Check if the given type is a valid base type to be used in access tags.
 static bool isValidBaseType(QualType QTy) {
-  if (QTy->isReferenceType())
-    return false;
```

Comment:
This looks like a functional change to me; can you explain it a bit more?

Reply:
This is case  #1 in the intro. Namely this is immediately followed by 
```
 if (const RecordType *TTy = QTy->getAs<RecordType>()) {
   .. do stuff
   return true;
}
return false;
```
So AFAICT that inital check for a reference type is doing nothing.  Am I misunderstanding getAs<RecordType> and that it's possible for a type to be both RecordType and ReferenceType?